ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -289,7 +295,66 @@ public void execute(Input input, ActionListener<Output> listener) {
                                                     listener.onFailure(ex);
                                                 }));
                                         } else {
-                                            executeAgent(inputDataSet, mlTask, isAsync, memoryId, mlAgent, outputs, modelTensors, listener);
+                                            // For existing conversations, create memory instance using factory
+                                            if (memorySpec != null && memorySpec.getType() != null) {
+                                                ConversationIndexMemory.Factory factory = (ConversationIndexMemory.Factory) memoryFactoryMap
+                                                    .get(memorySpec.getType());
+                                                if (factory != null) {
+                                                    // memoryId exists, so create returns an object with existing memory, therefore name can
+                                                    // be null
+                                                    factory
+                                                        .create(
+                                                            null,
+                                                            memoryId,
+                                                            appType,
+                                                            ActionListener
+                                                                .wrap(
+                                                                    createdMemory -> executeAgent(
+                                                                        inputDataSet,
+                                                                        mlTask,
+                                                                        isAsync,
+                                                                        memoryId,
+                                                                        mlAgent,
+                                                                        outputs,
+                                                                        modelTensors,
+                                                                        listener,
+                                                                        createdMemory
+                                                                    ),
+                                                                    ex -> {
+                                                                        log
+                                                                            .warn(
+                                                                                "Failed to create memory for existing conversation, proceeding without memory",
```

Comment:
why do we need to create memory for existing conversation?

Reply:
If customer provides a wrong memoryId, better to return error to let user know?